client/src/utils/fetch.js

```diff
@@ -67,18 +67,27 @@ async function _getAPI(uri) {
   }
 
   return data;
-}
+};
+
+export const getProduct = async productId => {
+  return _getAPI(`products/${productId}`);
+};
+
+export const getActiveProduct = async () => {
+  return _getAPI('active/product/');
+};
 
-async function _postAPI(uri, callback) {
+export const buyProduct = async (productId, callback) => {
+  let uri = `products/${productId}/purchase/`;
   const { API_URL } = getConfig();
 
   let url = `${API_URL}/${uri}`;
   try {
-    const csrfToken = _getAPI('csrf_token');
+    const token = await _getAPI('csrf_token');
 
     await fetch(url, {
       method: 'POST',
-      headers: { 'X-CSRFToken': csrfToken },
+      headers: { 'X-CSRFToken': token.csrfToken },
       ...baseRequest,
     });
     callback && callback(); // callbacks handle error message parsing directly.
```

Comment:
question: It seems like if callback is undefined, the UI would enter an inconsistent state. When would callback be undefined? Should an undefined callback instead prevent the buy API call from being sent?

This is likely outside current PR scope.

Reply:
https://github.com/GoogleCloudPlatform/avocano/issues/339